### PR TITLE
Ported DI from Unity to AutoFac.

### PR DESCRIPTION
**Pull request by galvesribeiro**
In order to be able to move Yams projects to .xproj so it cam cross-compile to .Net Framework and .Net Core, the DI container previously used, Unity, has to be replaced with AutoFac since the previously wasn`t supported on .Net Core.

- Ported DI from Unity to AutoFac.
- Added Portability Analysis report for both Yams and its tests.

### CODE REVIEW

File: src/Etg.Yams/DIModule.cs [hunk 1]
```diff
@@ -22,205 +22,193 @@ namespace Etg.Yams
...
+            RegisterDeploymentWatcher(builder);
 
-			RegisterDeploymentRepository(container);
+            RegisterDeploymentRepository(builder);
         }
 
-        private static void RegisterDeploymentRepository(IUnityContainer container)
+        private static void RegisterDeploymentRepository(ContainerBuilder builder)
         {
-            container.RegisterType<IDeploymentRepository>(new ContainerControlledLifetimeManager(), new InjectionFactory(
```
Comment (nehmebilal) [inline src/Etg.Yams/DIModule.cs]: Everywhere you see `ContainerControlledLifetimeManager` means that the class is being registered as a singleton. We need to do the same with Autofac.

File: test/Etg.Yams.Test/PortabilityAnalysis.html [hunk 2]
```diff
@@ -0,0 +1,528 @@
+<!--
```
Comment (nehmebilal) [inline test/Etg.Yams.Test/PortabilityAnalysis.html]: Is there a reason why the portability analysis need to be checked in?

File: src/Etg.Yams/YamsEntryPoint.cs [hunk 3]
```diff
@@ -12,24 +12,24 @@ public class YamsEntryPoint
...
-            _deploymentWatcher = unityContainer.Resolve<IDeploymentWatcher>();
-            _applicationPool = unityContainer.Resolve<IApplicationPool>();
-            _unityContainer = unityContainer;
+            _deploymentWatcher = container.Resolve<IDeploymentWatcher>();
+            _applicationPool = container.Resolve<IApplicationPool>();
+            _container = container;
         }
 
-        public static IUnityContainer InitializeDefaultUnityContainer(YamsConfig config)
+        public static IContainer InitializeDefaultUnityContainer(YamsConfig config)
```
Comment (nehmebilal) [inline src/Etg.Yams/YamsEntryPoint.cs]: You forgot the *Unity* here :)

`InitializeDefaultModules` is probably a better name for this method?

File: test/Etg.Yams.Test/EndToEndTest.cs [hunk 4]
```diff
@@ -34,14 +34,13 @@ public EndToEndTest()
...
-            DiModule.RegisterTypes(_diContainer, new YamsConfigBuilder("UseDevelopmentStorage=true", "deploymentId1", "1", "instanceId", _applicationsInstallPath).Build());
+            var builder = new ContainerBuilder();
+            DiModule.RegisterTypes(builder, new YamsConfigBuilder("UseDevelopmentStorage=true", "deploymentId1", "1", "instanceId", _applicationsInstallPath).Build());
 
             // Replace the IRemoteDirectory default implementation (which uses Azure) with a LocalDirectory implementation
             // so we can use local test data.
-            _diContainer.RegisterType<IDeploymentRepository>(new ContainerControlledLifetimeManager(),
-                new InjectionFactory(
-                    c => new LocalDeploymentRepository(_deploymentDirPath)));
+            builder.Register<IDeploymentRepository>(
```
Comment (nehmebilal) [inline test/Etg.Yams.Test/EndToEndTest.cs]: SingleInstance :)

File: src/Etg.Yams/DIModule.cs [hunk 5]
```diff
@@ -22,205 +22,193 @@ namespace Etg.Yams
     /// </summary>
```
Comment (nehmebilal) [inline src/Etg.Yams/DIModule.cs]: Would be great if you can delete the note above :)

**Comment by nehmebilal** [comment]
Portability reports are still there :)

**Comment by nehmebilal** [comment]
Also squashing commits using `git rebase -i HEAD~n` (`n` is the number of commits) would be great.
Once you have the commits squashed, you can push to your fork using the `-f` flag and it'll rewrite the branch without closing the PR :)

**Comment by nehmebilal** [comment]
LGTM!

**Comment by remusrg** [comment]
Same here, I'll merge. Thank you!
We'll need to circle back to the blob client as it looks like we can just have a singleton. Will open a new issue around that, should be quick check/fix.

**Pull request merged**